src/internal/tester/its/m365.go

```diff
@@ -0,0 +1,209 @@
+package its
+
+import (
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/h2non/gock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/alcionai/corso/src/internal/common/idname"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/internal/tester/tconfig"
+	"github.com/alcionai/corso/src/pkg/account"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/count"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+	"github.com/alcionai/corso/src/pkg/services/m365/api/graph"
+)
+
+// ---------------------------------------------------------------------------
+// Gockable client
+// ---------------------------------------------------------------------------
+
+// GockClient produces a new exchange api client that can be
+// mocked using gock.
+func gockClient(creds account.M365Config, counter *count.Bus) (api.Client, error) {
+	s, err := graph.NewGockService(creds, counter)
+	if err != nil {
+		return api.Client{}, err
+	}
+
+	li, err := graph.NewGockService(creds, counter, graph.NoTimeout())
+	if err != nil {
+		return api.Client{}, err
+	}
+
+	return api.Client{
+		Credentials: creds,
+		Stable:      s,
+		LargeItem:   li,
+	}, nil
+}
+
+// ---------------------------------------------------------------------------
+// Intercepting calls with Gock
+// ---------------------------------------------------------------------------
+
+const graphAPIHostURL = "https://graph.microsoft.com"
+
+func V1APIURLPath(parts ...string) string {
+	return strings.Join(append([]string{"/v1.0"}, parts...), "/")
+}
+
+func InterceptV1Path(pathParts ...string) *gock.Request {
+	return gock.New(graphAPIHostURL).Get(V1APIURLPath(pathParts...))
+}
+
+// ---------------------------------------------------------------------------
+// Suite Setup
+// ---------------------------------------------------------------------------
+
+type IDs struct {
+	Provider          idname.Provider
+	ID                string
+	Email             string
+	DriveID           string
+	DriveRootFolderID string
+	TestContainerID   string
+	WebURL            string
+}
+
+// m365IntgTestSetup provides all the common references used in an m365 integration
+// test suite.  Call `its.GetM365()` to get the singleton for your test suite.
+// If you're looking for unit testing setup, use `uts.GetM365()` instead.
+type m365IntgTestSetup struct {
```

Comment:
The unit and integration test setups seem to share a lot of the same fields and data. Why not have the integration test setup embed the unit test setup? That way the fields and tests for the test setup only need to be defined once

Reply:
Mostly to keep with "a little bit of duplication is better than a little bit of abstraction".  Honestly, though, I'm not sure I convinced myself of the value of the unit test variation by the time I finished plugging in the integration version.  I'm thinking about cutting it out entirely.  What are your feelings on it?